src/main/java/com/codicesoftware/plugins/hudson/PlasticSCM.java

```diff
@@ -239,17 +242,27 @@ private static boolean isSharedLibrary(@Nonnull FilePath jenkinsPath) {
         return jenkinsPath.getParent().getName().endsWith("@libs");
     }
 
+    /**
+     * Backward compatibility for jobs using obsolete configurations.
+     */
+    private void adoptOlderConfigurations() {
```

Comment:
I'd prefer not to change the object state in this method - it's kind of obscure to me. I'd rather have this code in the calling method (`checkout()`).

If you really want to have this isolated, you could rename the method to something like `updateCleanupMethodFromOlderConfig()` or similar so one doesn't need to go inside the method to find out what it does.

Reply:
I do not understand what you mean. This is the `checkout(...)` method which is the entry point for Jenkins SCM integrations. Where would you want it to be moved?

Regarding the name, my idea was to have a method that checks if job config is in obsolete format and applies it's settings to currently existing formula. I am aware that there is now just one field, the cleanup method, but the name was made with planning ahead, that there might be more fields like this.

Maybe leave this method and rename it to something more explanatory and do more verbose javadoc comment? `adjustConfigurationFromOlderFormat(...)` ?

Reply:
I think the new name is better indeed. Maybe we can improve this in the future, but it's good for the time being 😊